server/src/event/model/set_taxonomy_parent.rs

```diff
@@ -28,3 +31,61 @@ impl TryFrom<&AbstractEvent> for SetTaxonomyParentEvent {
         })
     }
 }
+
+pub struct SetTaxonomyParentEventPayload {
+    raw_typename: RawEventType,
+    child_id: i32,
+    previous_parent_id: Option<i64>,
```

Comment:
```suggestion
    previous_parent_id: i32,
```

We would create a loop when we move root -> should result in `BadRequest`